crates/shared/src/current_block.rs

```diff
@@ -124,6 +138,21 @@ where
     }
 }
 
+/// Determines if a new block is recent enough that it would warrant informing the system about it.
+/// This is used to protect against receiving very old blocks after the load balancer switched to a
+/// different node that is stuck in the past. The new block is allowed to be at most
+/// `MAX_REORG_BLOCK_COUNT` blocks older than the current block to still get notified about reorgs.
+fn updated_current_block_with_new_block(current_block: &Mutex<u64>, new_block: u64) -> bool {
+    let mut block_guard = current_block.lock().unwrap();
+    if new_block >= (*block_guard).saturating_sub(MAX_REORG_BLOCK_COUNT) {
+        // new block is within reorg depth -> update current block
+        *block_guard = new_block;
```

Comment:
I would change this logic.

I think that reorgs don't make the chain shorter, just rebase to a different chain (if that makes sense). This is definitely the case with PoW where you would go form:

```
a -> b -> c
```

to
```
a -> b' -> c' -> d
```

But not to:
```
a -> b
```

I think this is also the case for PoS.

So, I would change to:
- if block is newer
  - update latest block 🎉 
- if block is older
  - if block is more than `MAX_REORG_BLOCK_COUNT` older
    - alert
  - if block is less than `MAX_REORG_BLOCK_COUNT` older
    - ignore the block (maybe log a warning, increment a metric, etc.)

Reply:
Since both suspiscious cases (block very old and block very far ahead) seem to be very rare I decided to log errors in both cases instead of using metrics.

Reply:
Logging errors makes sense in that context for sure!